text/0059-otlp-trace-data-format.md

```diff
@@ -0,0 +1,346 @@
+  // then no events were dropped.
+  int32 dropped_timed_events_count = 13;
+
+  // Link is a pointer from the current span to another span in the same trace or in a
+  // different trace. For example, this can be used in batching operations,
+  // where a single batch handler processes multiple requests from different
+  // traces or when the handler receives a request from a different project.
+  message Link {
+    // trace_id is a unique identifier of a trace that this linked span is part of.
+    // The ID is a 16-byte array.
+    bytes trace_id = 1;
+
+    // span_id is a unique identifier for the linked span. The ID is an 8-byte array.
+    bytes span_id = 2;
+
+    // tracestate is the trace state associated with the link.
+    repeated TraceStateEntry tracestate = 3;
+
+    // attributes is a collection of attribute key/value pairs on the link.
+    repeated AttributeKeyValue attributes = 4;
+
+    // dropped_attributes_count is the number of dropped attributes. If the value is 0,
+    // then no attributes were dropped.
+    int32 dropped_attributes_count = 5;
+  }
+
+  // links is a collection of Links, which are references from this span to a span
+  // in the same or different trace.
+  repeated Link links = 14;
+
+  // dropped_links_count is the number of dropped links after the maximum size was
+  // enforced. If this value is 0, then no links were dropped.
+  int32 dropped_links_count = 15;
+
+  // status is an optional final status for this span. Semantically when status
+  // wasn't set it is means span ended without errors and assume Status.Ok (code = 0).
+  Status status = 16;
+
+  // child_span_count is an optional number of child spans that were generated while this
+  // span was active. If set, allows an implementation to detect missing child spans.
+  int32 child_span_count = 17;
+}
+
+// The `Status` type defines a logical error model that is suitable for different
+// programming environments, including REST APIs and RPC APIs.
+message Status {
+  // The status code. This is optional field. It is safe to assume 0 (OK)
+  // when not set.
+  int32 code = 1;
+
+  // A developer-facing error message, which should be in English.
+  string message = 2;
+}
+```
+
+### AttributeKeyValue
+
+```protobuf
+// AttributeKeyValue is a key-value pair that is used to store Span attributes, Resource
+// labels, etc.
+message AttributeKeyValue {
+  // ValueType is the enumeration of possible types that value can have.
+  enum ValueType {
+    STRING  = 0;
+    BOOL    = 1;
+    INT64   = 2;
+    DOUBLE  = 3;
+  };
+
+  // key part of the key-value pair.
+  string key = 1;
+
+  // The type of the value.
+  ValueType type = 2;
+
+  // Only one of the following fields is supposed to contain data (determined by `type` field value).
+  // This is deliberately not using Protobuf `oneof` for performance reasons (verified by benchmarks).
+
+  // A string up to 256 bytes long.
```

Comment:
this is a new limitation. Should it be also mentioned in specs?

Reply:
I copied this from OpenCensus proto, but I wonder if we need it in protobuf definition? If there is a need for limitation I think it can be in SDK, doesn't have to be here. I am willing to remove this and allow unlimited length strings. What do you think?

Reply:
go for it

Reply:
Length limitation removed.